d/extism.d

```diff
@@ -0,0 +1,258 @@
+module extism;
+
+import std.conv: castFrom, to;
+import std.meta: Alias;
+import std.string: fromStringz, toStringz;
+
+import runtime;
+
+/// A list of all possible value types in WebAssembly.
+enum ValType {
+  i32 = I32,
+  i64 = I64,
+  f32 = F32,
+  f64 = F64,
+  v128 = V128,
+  funcRef = FuncRef,
+  externRef = ExternRef,
+}
+
+// Opaque Pointers
+///
+alias CancelHandle = Alias!(void*);
+///
+alias CurrentPlugin = Alias!(void*);
+
+///
+alias ExtismSize = ulong;
+
+/// A union type for host function argument/return values.
+union ValUnion {
+  ///
+  int i32;
+  ///
+  long i64;
+  ///
+  float f32;
+  ///
+  double f64;
+}
+
+/// Holds the type and value of a function argument/return.
+struct Val {
+  ///
+  ValType t;
+  ///
+  ValUnion v;
+}
+
+/// Host function signature.
+///
+/// Used to register host functions with plugins.
+/// Params:
+///     plugin: the currently executing plugin from within a host function
+///     inputs: argument values
+///     outputs: return values
+///     data: user data associated with the host function
+alias FunctionType = void delegate(
+  CurrentPlugin plugin, const Val[] inputs, Val[] outputs, void* data
+);
+
+/// Returns: A slice of an allocated memory block of the currently running plugin.
+ubyte[] memory(CurrentPlugin plugin, ulong n) {
+  auto length = extism_current_plugin_memory_length(cast(ExtismCurrentPlugin*) plugin, n);
+  return extism_current_plugin_memory(cast(ExtismCurrentPlugin*) plugin)[n .. length];
```

Comment:
I think this should be:

```suggestion
  return extism_current_plugin_memory(cast(ExtismCurrentPlugin*) plugin)[n .. n + length];
```

I don't know the language very well, so sorry if that's incorrect. I'm just going through the code-path that would be executed and looking for any potential issues.